docs/source/java/flight.rst

```diff
@@ -0,0 +1,302 @@
+
+Writing a Flight Service
+========================
+
+Arrow Flight is a development framework to offer to us the building blocks
+to implement applications base on our needs.
+
+In this journey, you need to implement how do you are planning to expose
+your business operations in base of RPC Methods and Request Patterns
+provided by Flight.
+
+For Flight Server side operations you need to implement FlightProducer interface
+methods.
+
+.. code-block:: Java
+
+    public class TutorialFlightProducer implements FlightProducer {
+        @Override
+        // Override methods or use NoOpFlightProducer for only methods needed
+    }
+
+Also you need to specify where to listen for, let's combine producer and location
+to create and start the Flight Server. This will start the server, but won't block
+the rest of the program. Call `FlightServer.awaitTermination` to block until the
+server stops.
+
+.. code-block:: Java
+
+    import org.apache.arrow.flight.Action;
+    import org.apache.arrow.flight.ActionType;
+    import org.apache.arrow.flight.Criteria;
+    import org.apache.arrow.flight.FlightDescriptor;
+    import org.apache.arrow.flight.FlightInfo;
+    import org.apache.arrow.flight.FlightProducer;
+    import org.apache.arrow.flight.FlightServer;
+    import org.apache.arrow.flight.FlightStream;
+    import org.apache.arrow.flight.Location;
+    import org.apache.arrow.flight.PutResult;
+    import org.apache.arrow.flight.Result;
+    import org.apache.arrow.flight.SchemaResult;
+    import org.apache.arrow.flight.Ticket;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.util.AutoCloseables;
+
+    class TutorialFlightProducer implements FlightProducer {
+        @Override
+        // Override methods or use NoOpFlightProducer for only methods needed
+    }
+
+    Location location = Location.forGrpcInsecure("0.0.0.0", 0);
+
+    try(
+        BufferAllocator allocator = new RootAllocator();
+        FlightServer tutorialFlightServer = FlightServer.builder(
+                allocator,
+                location,
+                new TutorialFlightProducer()
+        ).build();
+    ){
+        tutorialFlightServer.start();
+        System.out.println("Server listening on port " + tutorialFlightServer.getPort());
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                System.out.println("\nExiting command...");
+                AutoCloseables.close(tutorialFlightServer, allocator);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }));
+        tutorialFlightServer.awaitTermination();
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+
+.. code-block:: shell
+
+    Server listening on port 58104
+
+Using the Flight Client
+=======================
+
+To connect to a Flight service, call `FlightClient.builder` with a location.
```

Comment:
Either use double-backticks or link to the javadocs.

Reply:
The same goes for below.

Reply:
Added java docs link